pkg/api/v1alpha1/controlplaneupgrade_types.go

```diff
@@ -0,0 +1,52 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ControlPlaneUpgradeSpec defines the desired state of ControlPlaneUpgrade.
+type ControlPlaneUpgradeSpec struct {
```

Comment:
I think we need documentation for every field on expected formats, expected objects that are being referenced, constraints etc.

Reply:
Same as above. 
This PR is mainly just adding scaffolding so we can have a place to go and start adding code.
We'll have follow up PRs where we are going to add field descriptions and stuff.
Does that sound good?